UefiCpuPkg/Library/MmUnblockMemoryLib/MmUnblockMemoryLib.c

```diff
@@ -0,0 +1,95 @@
+/** @file
+  The instance of MM Unblock Page Library.
+  This library provides an interface to request non-MMRAM pages to be mapped/unblocked
+  from inside MM environment.
+  For MM modules that need to access regions outside of MMRAMs, the agents that set up
+  these regions are responsible for invoking this API in order for these memory areas
+  to be accessed from inside MM.
+
+  Copyright (c) Microsoft Corporation.
+  Copyright (c) 2024, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+#include <Uefi.h>
+#include <Guid/UnblockRegion.h>
+#include <Ppi/MmCommunication.h>
+#include <Library/HobLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/PeiServicesLib.h>
+#include <Base.h>
+
+/**
+  This API provides a way to unblock certain data pages to be accessible inside MM environment.
+  For the original design, when the CommonBufferX is allocated, Smi handler can sense directly, there is no need to
+  register which is not allowed anymore.
+  The driver, named DriverX, needs to sense a piece of memory to communicate with the mm environment. There are two ways:
+      1. Use a fixed communication buffer, allocated by the PEI driver.
+      2. Register a piece of memory through unblock memory request. Smi handlers and drivers can communicate through it.
+       Once this memory is registered it calls the communication protocol which can transfer the buffer, named CommonBufferX to
+       CommDxe, can copy to fixed communication buffer.
+       Therefore, thus mmCore can shadow this buffer in preparation for communication.
+  Here in this function, it implement the item 2 in above description.

```

Comment:
The above description says lots of things - "Once this memory is registered it calls ...., can copy to fixed communication buffer".
This function does not implement the above.